src/formatDate.js

```diff
@@ -9,6 +9,27 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const data = {};
+  const newDateFormat = [];
+  const dateParts = date.split(fromFormat.slice(3));
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    data[fromFormat[i]] = dateParts[i];
+  }
+
+  if (!fromFormat.includes('YY')) {
```

Comment:
This condition checks if 'YY' is not included in the `fromFormat`, but it does not handle the case where 'YYYY' is not included in the `fromFormat` but 'YY' is. This will lead to an incorrect assignment in line 21.